pkg/engine/task/task_apply.go

```diff
@@ -99,33 +137,87 @@ func apply(rr []runtime.Object, c client.Client, di discovery.DiscoveryInterface
 	return applied, nil
 }
 
-// patch calls update method on kubernetes client to make sure the current resource reflects what is on server
-//
-// an obvious optimization here would be to not patch when objects are the same, however that is not easy
-// kubernetes native objects might be a problem because we cannot just compare the spec as the spec might have extra fields
-// and those extra fields are set by some kubernetes component
-// because of that for now we just try to apply the patch every time
-// it mutates the object passed in to be consistent with the kubernetes client behavior
-func patch(newObj runtime.Object, c client.Client) error {
-	newObjJSON, _ := apijson.Marshal(newObj)
-	key, _ := client.ObjectKeyFromObject(newObj)
+func patch(updatedObj, currentObj runtime.Object, c client.Client) error {
```

Comment:
In the spirit of keeping the terminology consistent with the `threeWayMegerPatch(original, modified, current)`method, let's `s/updatedObj/modifiedObj`